src/databricks/labs/ucx/source_code/graph.py

```diff
@@ -489,3 +497,57 @@ class MaybeGraph:
     @property
     def failed(self):
         return len(self.problems) > 0
+
+
+class InheritedContext:
```

Comment:
isn't it redundant to be returned from public api (` -> InheritedContext:`)? it's essentially `(Tree | None, bool)`, where we only have `(None, False)` and `(Tree(...), True)`. Do we have `(None, True)` or `(Tree(...), False)` as well?

Reply:
it's fine to have `InheritedContext`, don't get me wrong - but my question is - shouldn't we return `Tree | None` only in public methods.

Reply:
I see what you mean, I'll think about it

Reply:
I've reduced the footprint slightly in commit 1613e75b98a26669800494cc5455a6b62e42c02b